core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -165,6 +168,13 @@ object SparkSubmit {
       }
     }
 
+    // In yarn-cluster mode for a python app, add primary Resource and pyFiles to files
+    // that can be distributed with the job
+    if (args.isPython && deployMode == CLUSTER) {
```

Comment:
i will add clusterManager == YARN && deployMode == CLUSTER that can says yarn-cluster mode.